pkg/testutil/scenarios.go

```diff
@@ -42,14 +57,23 @@ type PathContentFlag struct {
 	Content func() string
 }
 
+// NewEnvConfig provides the details required to setup a temporary test
+// environment, and optionally a function to run which accepts the
+// environment directory and can modify fields in the TestScenario
+type NewEnvConfig struct {
+	EnvOpts      *EnvOpts
+	EditScenario func(*TestScenario, string)
+}
+
 // RunScenario executes a TestScenario struct.
-// The Args field of the scenario is prepended with the content of the 'command'
+// The Arg field of the scenario is prepended with the content of the 'command'
```

Comment:
I'm not a fan of `Arg` (singular) as it suggests a _single_ argument, rather than potentially numerous flags/args within the string itself (e.g. `Arg:  "-s 123 --version 1 --package pkg/package.tar.gz --autoclone",`).

I understand it was named this way because `Args` was already used. But I would suggest we switch from `Arg` to `Args` once the latter is no longer used within the CLI code base.

Reply:
Great minds think alike... I had planned to do just that, and also contemplated using something a bit 'random' for the temporary field name, but decided not to because it wasn't going to be replaced within a day or two.